api/v1beta1/placementapi_webhook.go

```diff
@@ -86,7 +92,27 @@ func (r *PlacementAPI) ValidateCreate() error {
 func (r *PlacementAPI) ValidateUpdate(old runtime.Object) error {
 	placementapilog.Info("validate update", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object update.
+	oldPlacement, ok := old.(*PlacementAPI)
+	if !ok || oldPlacement == nil {
+		return apierrors.NewInternalError(fmt.Errorf("unable to convert existing object"))
+	}
+
+	if r.Spec.DatabaseInstance != oldPlacement.Spec.DatabaseInstance {
+		if r.Spec.Immutable.DatabaseInstance {
+			return apierrors.NewForbidden(
+				schema.GroupResource{
+					Group:    GroupVersion.WithKind("PlacementAPI").Group,
+					Resource: GroupVersion.WithKind("PlacementAPI").Kind,
+				}, r.GetName(), &field.Error{
+					Type:     field.ErrorTypeForbidden,
+					Field:    "*",
```

Comment:
I think we should report the which field we are reporting the error on here. E.g. this should be "spec.databaseInstance" if the local webhook validated it, or "spec.placement.template.databaseInstance" if the openstack-operator validated it. You can pass a field.Path here to get the necessary base path of the field for the two different call path. See https://github.com/openstack-k8s-operators/nova-operator/pull/411/files#diff-0bb4bc5fee241020d547ea9eb775efb27b4d8082e09ad7d51a221cebb8cd6002R110 for example

Reply:
Fixed in the latest version